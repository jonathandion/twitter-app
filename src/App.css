.App {
    text-align: center;
    max-width: 1280px;
    margin: 0 auto;
    padding: 50px 0;
}