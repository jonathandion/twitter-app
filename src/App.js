// @flow

import React, {Component} from 'react';
import {connect} from 'react-redux';
import {Spin, Alert} from 'antd';

import {fetchTweets} from './actions/index.actions';

import Tweets from './components/Tweets/Tweets.component';
import Nav from './components/Nav/Nav.component';
import Hero from './components/Hero/Hero.component';

import './App.css';

interface TwitterUser {
  fullname: string,
  username: string,
}

const twitterUsers: Array<TwitterUser> = [
  {
    fullname: 'Donald Trump',
    username: 'realDonaldTrump',
  },
  {
    fullname: 'Hillary Clinton',
    username: 'hillaryclinton',
  },
  {
    fullname: 'Frank Underwood',
    username: 'Frank_Underwood',
  },
];

export class App extends Component {
  state: {
    selected: string,
  } = {
    selected: 'realDonaldTrump',
  };
  componentDidMount() {
    this.fetchTweets(this.state.selected);
  }

  fetchTweets = (value: string) => {
    this.props.dispatch(fetchTweets(value));
    this.setState({selected: value});
  };

  render() {
    const {tweets = [], isFetching, errors} = this.props;

    return (
      <main className="App">
        <Hero user={this.state.selected} tweets={tweets} />
        <div className="o-container">
          <Nav
            fetchTweets={this.fetchTweets}
            selected={this.state.selected}
            users={twitterUsers}
          />
          {renderTweets({data: tweets, errors, isFetching})}
        </div>
      </main>
    );
  }
}

function renderTweets({
  data,
  errors,
  isFetching,
}: {
  data: Array<Object>,
  errors: string,
  isFetching: Boolean,
}) {
  if (isFetching) {
    return <Spin />;
  } else if (errors) {
    return (
      <Alert
        message="Error"
        description="A problem occured"
        type="error"
        showIcon
      />
    );
  } else if (data.length === 0 && isFetching === false) {
    return (
      <Alert
        message="Warning"
        description="No tweets :("
        type="warning"
        showIcon
      />
    );
  }

  return <Tweets data={data} />;
}

const mapStateToProps = (state: Object) => {
  return {
    ...state.tweets,
  };
};

export default connect(mapStateToProps)(App);
