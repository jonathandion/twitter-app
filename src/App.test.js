import React from 'react';
import ReactDOM from 'react-dom';
import {App} from './App';
import renderer from 'react-test-renderer';

describe('<App />', function() {
  it('renders correctly', function() {
    const dispatch = jest.fn();
    const props = {
      dispatch,
      tweets : []
    };
    const tree = renderer.create(<App {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
