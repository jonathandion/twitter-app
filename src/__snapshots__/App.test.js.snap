// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components <App /> renders correctly 1`] = `
<div
  className="App"
>
  <header>
    <div
      className="ant-radio-group"
      onMouseEnter={undefined}
      onMouseLeave={undefined}
      style={undefined}
    >
      <label
        className="ant-radio-button-wrapper ant-radio-button-wrapper-checked"
        onMouseEnter={undefined}
        onMouseLeave={undefined}
        style={undefined}
      >
        <span
          className="ant-radio-button ant-radio-button-checked"
          style={Object {}}
        >
          <input
            checked={true}
            className="ant-radio-button-input"
            disabled={false}
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={undefined}
            onFocus={[Function]}
            readOnly={undefined}
            tabIndex={undefined}
            type="radio"
          />
          <span
            className="ant-radio-button-inner"
          />
        </span>
        <span>
          Donald Trump
        </span>
      </label>
      <label
        className="ant-radio-button-wrapper"
        onMouseEnter={undefined}
        onMouseLeave={undefined}
        style={undefined}
      >
        <span
          className="ant-radio-button"
          style={Object {}}
        >
          <input
            checked={false}
            className="ant-radio-button-input"
            disabled={false}
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={undefined}
            onFocus={[Function]}
            readOnly={undefined}
            tabIndex={undefined}
            type="radio"
          />
          <span
            className="ant-radio-button-inner"
          />
        </span>
        <span>
          Hillary Clinton
        </span>
      </label>
    </div>
    <span
      className="ant-input-affix-wrapper"
      style={
        Object {
          "width": 200,
        }
      }
    >
      <input
        className="ant-input ant-input-search"
        disabled={false}
        onKeyDown={[Function]}
        placeholder="Search username"
        style={null}
        type="text"
      />
      <span
        className="ant-input-suffix"
      >
        <i
          className="anticon anticon-search ant-input-search-icon"
          onClick={[Function]}
        />
      </span>
    </span>
  </header>
  <div
    className="c-cards"
  />
</div>
`;
